common-pg/src/main/scala/com/socrata/pg/store/PGSecondaryDatasetMapReader.scala

```diff
@@ -1,53 +1,174 @@
+       dataset_map.system_id           as dataset_map_system_id,
+       dataset_map.next_counter_value  as dataset_map_next_counter_value,
+       dataset_map.latest_data_version as dataset_map_latest_data_version,
+       dataset_map.locale_name         as dataset_map_locale_name,
+       dataset_map.obfuscation_key     as dataset_map_obfuscation_key,
+       dataset_map.resource_name       as dataset_map_resource_name,
+       copy_map.system_id as copy_map_system_id,
+       copy_map.copy_number as copy_map_copy_number,
+       copy_map.lifecycle_stage :: TEXT as copy_map_lifecycle_stage,
+       copy_map.data_version as copy_map_data_version,
+       copy_map.data_shape_version as copy_map_data_shape_version,
+       copy_map.last_modified as copy_map_last_modified,
+       copy_map_table_modifiers.table_modifier as copy_map_table_modifiers_table_modifier
+from rollup_relationship_map
+         join rollup_map on rollup_relationship_map.rollup_system_id = rollup_map.system_id
+         join copy_map on rollup_map.copy_system_id = copy_map.system_id
+         left join copy_map_table_modifiers ON copy_map.system_id = copy_map_table_modifiers.copy_system_id
+         join dataset_map on copy_map.dataset_system_id = dataset_map.system_id
+where referenced_copy_system_id = ?"""
+
+  def getRollupsRelatedToCopy(copyInfo: CopyInfo): Set[LocalRollupInfo] = {
+    using(conn.prepareStatement(getRollupsRelatedToCopyQuery)) { stmt =>
+      stmt.setLong(1, copyInfo.systemId.underlying)
+      using(t("getRollupsRelatedToCopy", "copy_id" -> copyInfo.systemId)(stmt.executeQuery())) { rs =>
+        Iterator.continually(rs.next())
+          .takeWhile(identity)
+          .map(_=>
+            new LocalRollupInfo(
+              CopyInfo(
+                DatasetInfo(new DatasetId(rs.getLong("dataset_map_system_id")), rs.getLong("dataset_map_next_counter_value"), rs.getLong("dataset_map_latest_data_version"), rs.getString("dataset_map_locale_name"), rs.getBytes("dataset_map_obfuscation_key"), Option(rs.getString("dataset_map_resource_name"))),
+                new CopyId(rs.getLong("copy_map_system_id")),
+                rs.getLong("copy_map_copy_number"),
+                LifecycleStage.valueOf(rs.getString("copy_map_lifecycle_stage")),
+                rs.getLong("copy_map_data_version"),
+                rs.getLong("copy_map_data_shape_version"),
+                new DateTime(rs.getTimestamp("copy_map_last_modified").getTime),
+                Some(rs.getLong("copy_map_table_modifiers_table_modifier"))
+              ),
+              new RollupName(rs.getString("rollup_map_name")),
+              rs.getString("rollup_map_soql"),
+              rs.getString("rollup_map_table_name"),
+              RollupId(rs.getLong("rollup_map_system_id")))
+          ).toSet
+      }
+    }
+  }
+
+  def nextSequenceValue = "select nextval(?)"
+  def getNextSequence[T](sequenceName: String): Option[T] = {
```

Comment:
This type is a lie; it says "I can give you any type T that you ask for".

But sequences always return that are smaller than `bigint`, so this T must (a) always be such an integer, and (b) can always losslessly be `Long`.

Reply:
s/smaller than/no larger than/

Reply:
Oh! True, let me fix that up.
But correct my intent was to allow any of the sequence types.